lib/_http_client.js

```diff
@@ -562,7 +562,13 @@ function tickOnSocket(req, socket) {
   socket.on('close', socketCloseListener);
 
   if (req.timeout) {
-    socket.once('timeout', () => req.emit('timeout'));
+    const emitRequestTimeout = () => req.emit('timeout');
```

Comment:
Not to derail this PR which is desperately needed, but isn't this section very similar to the existing `ClientRequest.prototype.setTimeout` method?    Would just calling `req.setTimeout(req.timeout)` at this point achieve almost the same behavior?